walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,198 @@
+      ArrayList<TimeRange> availableTimes) {
+    if (start > prevEnd) {
+      int availableDuration = start - prevEnd;
+      if (availableDuration >= duration) {
+        TimeRange availableTime = TimeRange.fromStartDuration(prevEnd, 
+            availableDuration);
+        availableTimes.add(availableTime);
+      }
+    }
+  }
+
+  /**
+   * Returns the times available to have the meeting of a certain duration
+   * Time Complexity: O(n)
+   *
+   * @param times Contains an arraylist of an array of start and end times
+   *    this array should be ordered and contain no duplicate start times
+   * @param duration the duration of time the meeting request lasts
+   * @return an arraylist of times available whose duration are >= duration
+   */
+  public static Collection<TimeRange> timeAvailable(ArrayList<int[]> times, 
+      int duration) {
+    int endTime = TimeRange.START_OF_DAY;
+    ArrayList<TimeRange> availableTimes = new ArrayList<TimeRange>();
+    for (int[] time : times) {
+      int start = time[0];
+      int end = time[1];
+      
+      addAvailableTime(start, endTime, duration, availableTimes);
+
+      endTime = end;
+    }
+
+    int endOfDay = TimeRange.END_OF_DAY + 1;
+    addAvailableTime(endOfDay, endTime, duration, availableTimes);
+
+    return availableTimes;
+  }
+  
+  /**
+   * Returns an array with all of the events that meet the requirements set in 
+   *    the predicate.
+   * Time Complexity: O(n^2 * ln(n))
+   *
+   * @param events the array of events that we are going to filter through
+   * @param pred the predicate that will decide whether or not to keep an event
+   * @retunr an array of events that meet the requirements set in the predicate
+   */
+  public static Event[] includeEventIf(Event[] events, Predicate<Event> pred) {
+    ArrayList<Event> filteredEvents = new ArrayList<Event>();
+    for (Event event : events) {
+      // if the events share an attenddee then add it to the array
+      if (pred.test(event)) {
+        filteredEvents.add(event);
+      }
+    }
+    Event[] filteredEventsArray = new Event[filteredEvents.size()];
+    filteredEventsArray = filteredEvents.toArray(filteredEventsArray);
+    return filteredEventsArray;
+  }
+
+  /**
+   * Returns all possible time periods throughout the day when everybody 
+   *    attending this meeting is available. 
+   * Time Complexity: O(n^2 * ln(n))
+   * 
+   * @param events All events that are occurring  
+   * @param request The meeting that the user wants to create and find time for 
+   * @return an array of TimeRange objects
+   */
+  public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) throws Exception{
+    // throw new UnsupportedOperationException("TODO: Implement this method.");
+
+    // check if duration of meeting is longer than a day or a negative number
+    int durationMeeting = Math.toIntExact(request.getDuration());
```

Comment:
What unit is this duration in? Add that to the variable name. For example, if it's minutes, rename: `durationMeetingMinutes` etc.